url2f44efcfa6_wangyudong_sohutw_java_sdk_2_0_tgz-pJ8-com_sohu_t_api_sdk_json_TestJ8/src/com/sohu/t/api/sdk/Configuration.java

```diff
@@ -0,0 +1,317 @@
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+package com.sohu.t.api.sdk;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.security.AccessControlException;
+import java.util.Properties;
+
+/**
+ * @author Yusuke Yamamoto - yusuke at mac.com
+ * 
+ * modified by chenxj
+ */
+public class Configuration {
+
+	private static Properties defaultProperty;
+
+    static {
+        init();
+    }
+
+    /**
+     * 初始化配置信息，先读取系统环境变量, tblog4j.properties会覆盖系统变量，
+     * classpath:/sdk.properties，WEB-INF/sdk.properties
+     */
+    static void init() {
+        defaultProperty = new Properties();
+        defaultProperty.setProperty("tblog4j.debug", "true");
+        defaultProperty.setProperty("tblog4j.clientVersion","");
+        defaultProperty.setProperty("tblog4j.clientURL", "http://open.t.sohu.com");
+        defaultProperty.setProperty("tblog4j.http.userAgent", "tblog4j http://open.t.sohu.com");
+
+        defaultProperty.setProperty("tblog4j.http.useSSL", "false");
+        defaultProperty.setProperty("tblog4j.http.proxyHost.fallback", "http.proxyHost");
+        defaultProperty.setProperty("tblog4j.http.proxyPort.fallback", "http.proxyPort");
+        defaultProperty.setProperty("tblog4j.http.connectionTimeout", "20000");
+        defaultProperty.setProperty("tblog4j.http.readTimeout", "120000");
+        defaultProperty.setProperty("tblog4j.http.retryCount", "3");
+        defaultProperty.setProperty("tblog4j.http.retryIntervalSecs", "10");
+        defaultProperty.setProperty("tblog4j.async.numThreads", "3");
+        defaultProperty.setProperty("tblog4j.clientVersion", Version.getVersion());
+        
+        try {
+            // Android platform should have dalvik.system.VMRuntime in the classpath.
+            // @see http://developer.android.com/reference/dalvik/system/VMRuntime.html
+            Class.forName("dalvik.system.VMRuntime");
+            defaultProperty.setProperty("tblog4j.dalvik", "true");
+        } catch (ClassNotFoundException cnfe) {
+            defaultProperty.setProperty("tblog4j.dalvik", "false");
+        }
+        DALVIK = getBoolean("tblog4j.dalvik");
+        String sdkProps = "sdk.properties";
+        boolean loaded = loadProperties(defaultProperty, "." + File.separatorChar + sdkProps) ||
+                loadProperties(defaultProperty, Configuration.class.getResourceAsStream("/WEB-INF/" + sdkProps)) ||
+                loadProperties(defaultProperty, Configuration.class.getResourceAsStream("/" + sdkProps));
+    }
+
+    private static boolean loadProperties(Properties props, String path) {
+        try {
+            File file = new File(path);
+            if(file.exists() && file.isFile()){
+                props.load(new FileInputStream(file));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).